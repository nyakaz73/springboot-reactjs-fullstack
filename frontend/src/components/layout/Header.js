import React from 'react';

function Header() {
    return (
        <header style={header} className="header">
            <h2>User Registration</h2>
        </header>
    )
}

const header = {
    display: "flex",
    justifyContent: "center",
    padding: '10px',
    color: "#fff"
}

export default Header;

