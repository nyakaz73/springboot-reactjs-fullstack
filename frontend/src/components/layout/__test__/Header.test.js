import React from "react";
import { cleanup, render } from "@testing-library/react";
import Header from "../Header";
//for Snapshot testing
import renderer from 'react-test-renderer';

//unmount and clean DOM after every test
afterEach(cleanup);

it("renders Header without crashing", () => {
    //Paint the DOM with component
    const { container } = render(<Header />)
    expect(container.textContent).toBe("User Registration")
})

it("renders <Header/> with a prop", () => {
    const title = "My Header"
    const { getByText } = render(<Header title={title} />)
    expect(getByText('My Header')).toBeTruthy()
})


it("matches snapshot", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
})
