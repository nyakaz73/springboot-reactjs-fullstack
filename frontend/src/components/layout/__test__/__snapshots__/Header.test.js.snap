// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header"
  style={
    Object {
      "color": "#fff",
      "display": "flex",
      "justifyContent": "center",
      "padding": "10px",
    }
  }
>
  <h2>
    User Registration
  </h2>
</header>
`;
